app/scripts/views/sign_up.js

```diff
@@ -48,14 +48,17 @@ function (Cocktail, _, p, BaseView, FormView, Template, AuthErrors, mailcheck,
     },
 
     beforeRender: function () {
+      var self = this;
       if (document.cookie.indexOf('tooyoung') > -1) {
         this.navigate('cannot_create_account');
         return p(false);
       }
 
       this._bouncedEmail = this.ephemeralMessages.get('bouncedEmail');
-
-      return FormView.prototype.beforeRender.call(this);
+      require(['passwordcheck'], function (PasswordCheck) {
```

Comment:
So who knows how to force `passwordcheck` to be loaded asynchronously? Can it be done with requirejs, or will we have to use some other mechanism? We'll have to correctly deal with caching. Since this is the signup page and most users only sign up once per device (huge assumption there), perhaps using an etag is sufficient and strong caching with cache busting is not needed.